hbase-server/src/main/java/org/apache/hadoop/hbase/master/AssignmentManager.java

```diff
@@ -2564,6 +2588,45 @@ public int compare(Pair<ServerName, String> o1, Pair<ServerName, String> o2) {
     return res;
   }
 
+  /**
+   * Get a list of servers that this region can not assign to.
+   * For system table, we must assign them to a server with highest version.
+   * This method is same as {@link #getExcludedServersForSystemTable()} with
+   * the only difference is we can disable this exclusion using config:
+   * "hbase.min.version.move.system.tables".
+   *
+   * @return List of Excluded servers for System table regions.
+   */
+  private List<ServerName> getExcludedServersForSystemTableUnlessAllowed() {
```

Comment:
Can we club the logic of both the methods and pass an optional boolean for version check?
The two functions that look very much alike except for the tail part.